### PR TITLE
statistics: reduce unnecessary copy when collect hot metrics

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/5692
Before:
image
After:
image

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note
```
None.
```

### CODE REVIEW

**Comment by rleungx** [comment]
Is there any benchmark to show the improvement?

**Comment by lhy1024** [comment]
> Is there any benchmark to show the improvement?

As shown in the picture in the description of pr, the memory of clone will be saved when collecting monitoring information, and in the new version of the test, the relevant flame graph will not be seen.

File: server/cluster/coordinator_test.go [hunk 1]
```diff
@@ -269,6 +269,47 @@ func TestCollectMetrics(t *testing.T) {
...
+	count := 10
+	for i := 0; i <= count; i++ {
+		for k := 0; k < 200; k++ {
+			item := &statistics.HotPeerStat{
+				StoreID:  uint64(i % 5),
+				RegionID: uint64(i*1000 + k),
+				Kind:     statistics.Write,
+				Loads:    []float64{10, 20, 30},
+			}
+			item.Warm()
```
Comment (rleungx) [inline server/cluster/coordinator_test.go:288]: Why not directly set the value?

File: server/statistics/store_hot_peers_infos.go [hunk 2]
```diff
@@ -32,6 +32,41 @@ type StoreHotPeersInfos struct {
...
+			for i := range peerLoadSum {
+				peerLoadSum[i] = 0
+			}
+			peers := filterHotPeers(kind, hotPeers)
+			for _, peer := range peers {
+				for j := range peerLoadSum {
+					peerLoadSum[j] += peer.GetLoad(j)
+				}
+			}
+			ret[id] = &HotPeersStat{
```
Comment (rleungx) [inline server/statistics/store_hot_peers_infos.go:55]: Will it miss some info compared with before?

File: server/cluster/coordinator_test.go (hunk 1 above)
Comment (lhy1024) [reply server/cluster/coordinator_test.go:288]: `hot-degree` and `anticount` is private

File: server/statistics/store_hot_peers_infos.go (hunk 2 above)
Comment (lhy1024) [reply server/statistics/store_hot_peers_infos.go:55]: It doesn't contain store info and hot peers detail, but metrics only use the total data of hot peers.

**Review by rleungx — APPROVED** [review]

**Comment by nolouch** [comment]
PTAL @HunDunDM

File: server/statistics/store_hot_peers_infos.go [hunk 3]
```diff
@@ -32,6 +32,41 @@ type StoreHotPeersInfos struct {
...
+			if !ok {
+				continue
+			}
+			for i := range peerLoadSum {
+				peerLoadSum[i] = 0
+			}
+			peers := filterHotPeers(kind, hotPeers)
+			for _, peer := range peers {
+				for j := range peerLoadSum {
+					peerLoadSum[j] += peer.GetLoad(j)
```
Comment (CabinfeverB) [inline server/statistics/store_hot_peers_infos.go:52]: In `summaryStoresLoadByEngine`, it seems use `peer.Loads[i]` but not `peer.GetLoad(j)`

**Review by CabinfeverB — COMMENTED** [review]
Rest LGTM

File: server/statistics/store_hot_peers_infos.go (hunk 3 above)
Comment (lhy1024) [reply server/statistics/store_hot_peers_infos.go:52]: `Peer.Clone` put denoising data to `Loads`, in other words, `peer.Loads[i]` and `peer.GetLoad(i)` are equal, and they both return denoising data. So we should use `peer.GetLoad(j)` in the new function.

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**